### PR TITLE
DT-1334 Capitalise POMs names and find existing Staff regardless of case

### PR DESCRIPTION
**Pull request by andymarke**
OMiC will supply POM names in upper case; e.g JOHN SMITH

This resolves two issues:
1) When Staff already exists regardless of case - use that existing staff, so JOHN SMITH will match an existing john smith
2) When creating a new staff member capitalise - e.g transform to John Smith

### CODE REVIEW

File: src/main/resources/db/schema/V1_0__schema.sql [hunk 1]
```diff
@@ -96,7 +96,7 @@ create sequence SC_PROVIDER_ID_SEQ
     ;
 
 create sequence STAFF_ID_SEQ
-    minvalue 2500000000
+    minvalue 2600000000
```
Comment (andymarke) [inline src/main/resources/db/schema/V1_0__schema.sql:99]: boost index values in test data so new rows to have duplicate values

File: src/test/java/uk/gov/justice/digital/delius/service/OffenderManagerService_allocatePrisonOffenderManagerTest.java [hunk 2]
```diff
@@ -80,502 +82,588 @@ public void setup() {
...
+        @Nested
+        @DisplayName("When successful")
+        class OnSuccess {
+            @BeforeEach
+            void setUp() {
+                when(staffService.findOrCreateStaffInArea(any(), any())).thenReturn(aStaff());
+            }
+
+            @Test
+            public void willCapitaliseStaffName() {
```
Comment (andymarke) [inline src/test/java/uk/gov/justice/digital/delius/service/OffenderManagerService_allocatePrisonOffenderManagerTest.java:648]: NEW TEST HERE -->>>>

File: src/test/java/uk/gov/justice/digital/delius/service/OffenderManagerService_allocatePrisonOffenderManagerTest.java [hunk 3]
```diff
@@ -80,502 +82,588 @@ public void setup() {
                 referenceDataService,
```
Comment (andymarke) [inline src/test/java/uk/gov/justice/digital/delius/service/OffenderManagerService_allocatePrisonOffenderManagerTest.java:82]: I have only added one test here, the huge change is simple due to moving all tests into a `@Nested` class for two main service methods.  

Look for new test down the page.

**Review by chris-faulkner-mojdt — APPROVED** [review]
LGTM

**Pull request merged**